deployment/environment/memory/node_service_client.go

```diff
@@ -71,24 +85,24 @@ func (j JobClient) GetNode(ctx context.Context, in *nodev1.GetNodeRequest, opts
 			PublicKey:   n.Keys.CSA.PublicKeyString(),
 			IsEnabled:   true,
 			IsConnected: true,
+			Labels:      n.Labels,
 		},
 	}, nil
 }
 
 func (j JobClient) ListNodes(ctx context.Context, in *nodev1.ListNodesRequest, opts ...grpc.CallOption) (*nodev1.ListNodesResponse, error) {
 	var nodes []*nodev1.Node
 	for id, n := range j.nodeStore.asMap() {
+		p2pIDLabel := &ptypes.Label{
+			Key:   "p2p_id",
+			Value: ptr(n.Keys.PeerID.String()),
+		}
```

Comment:
where is it getting used? This differs from usual implementation of nops or devnodes. Is this because you need peerIds from memory node and there is no other way to get it?

Reply:
I don't know, to be honest. This was in the previous implementation, so I kept it.